adapters/deeplabv3/model_adapter.py

```diff
@@ -0,0 +1,480 @@
+                                              filters=dl.Filters(custom_filter=train_filter),
+                                              dataset_entity=self.model_entity.dataset,
+                                              id_to_label_map=self.model_entity.id_to_label_map,
+                                              label_to_id_map=self.model_entity.label_to_id_map,
+                                              overwrite=False,
+                                              to_mask=False,
+                                              annotation_type=dl.AnnotationType.POLYGON,
+                                              transforms=data_transforms['train']
+                                              )
+
+        val_dataset = DatasetGeneratorTorch(data_path=os.path.join(data_path, 'validation'),
+                                            filters=dl.Filters(custom_filter=val_filter),
+                                            dataset_entity=self.model_entity.dataset,
+                                            id_to_label_map=self.model_entity.id_to_label_map,
+                                            label_to_id_map=self.model_entity.label_to_id_map,
+                                            overwrite=False,
+                                            to_mask=False,
+                                            annotation_type=dl.AnnotationType.POLYGON,
+                                            transforms=data_transforms['val']
+                                            )
+
+        dataloaders = {'train': DataLoader(train_dataset,
+                                           batch_size=batch_size,
+                                           shuffle=True,
+                                           collate_fn=self.dl_collate),
+                       'val': DataLoader(val_dataset,
+                                         batch_size=batch_size,
+                                         shuffle=True,
+                                         collate_fn=self.dl_collate,
+                                         )}
+        logger.debug("Train and Val data loaders created")
+
+        #####################
+        # Prepare the model #
+        #####################
+
+        n_classes = len(train_dataset.id_to_label_map)
+        logger.info('Setting last layer for {} classes'.format(n_classes))
+        self.model.classifier[-1] = torch.nn.Conv2d(256, n_classes, kernel_size=(1, 1), stride=(1, 1))
+
+        criterion = torch.nn.CrossEntropyLoss()  # For multi-class segmentation
+        # Observe that all parameters are being optimized
+        optimizer = optim.SGD(self.model.parameters(), lr=0.001, momentum=0.9)
+        # Decay LR by a factor of 0.1 every 7 epochs
+        exp_lr_scheduler = optim.lr_scheduler.StepLR(optimizer, step_size=3, gamma=0.1)
+
+        since = time.time()
+
+        # early stopping params
+        best_loss = np.inf
+        best_acc = 0.0
+        not_improving_epochs = 0
```

Comment:
Shouldn't these parameters also be available for the user in config?

Reply:
there is a patience_epochs parameter in the config